cpp/src/gis/perimeter.cu

```diff
@@ -0,0 +1,233 @@
+                        static_cast<col_type*>(polygons_longitudes[0]->data), polygons_latitudes[0]->size, lenghts_polygon.data().get() );
+
+            // Add all lenghts
+            col_type perimeter = thrust::reduce(lenghts_polygon.begin(), lenghts_polygon.end());
+            CUDA_TRY( cudaMemcpy(buffer_perimeter, &perimeter, sizeof(col_type) * num_polygons, cudaMemcpyHostToDevice) );
+            gdf_column_view(&perimeter_output, buffer_perimeter, nullptr, num_polygons, GDF_FLOAT64);
+
+        } else
+        {
+            gdf_column_view(&perimeter_output, buffer_perimeter, nullptr, num_polygons, GDF_FLOAT64);
+
+            // Concatenate latitudes of the polygons to one column (same for longitudes)
+            gdf_column* concat_latitudes = new gdf_column;
+            gdf_column* concat_longitudes = new gdf_column;
+
+            col_type* buffer_all_latitudes, *buffer_all_longitudes;
+            RMM_TRY( RMM_ALLOC((void**)&buffer_all_latitudes, sizeof(col_type) * total_points_of_all_polygons, 0) );
+            RMM_TRY( RMM_ALLOC((void**)&buffer_all_longitudes, sizeof(col_type) * total_points_of_all_polygons, 0) );
+
+            gdf_column_view(concat_latitudes, buffer_all_latitudes, nullptr, total_points_of_all_polygons, GDF_FLOAT64);
+            gdf_column_view(concat_longitudes, buffer_all_longitudes, nullptr, total_points_of_all_polygons, GDF_FLOAT64);
+
+            gdf_column_concat(concat_latitudes, polygons_latitudes, num_polygons);
+            gdf_column_concat(concat_longitudes, polygons_longitudes, num_polygons);
+
+            // Launch the optimized Kernel
+            CUDA_TRY( cudaOccupancyMaxPotentialBlockSize(&min_grid_size, &block_size, perimeters_kernel<col_type>) );
+            cudf::util::cuda::grid_config_1d grid{concat_latitudes->size, block_size, 1};
+
+            perimeters_kernel<col_type> <<< grid.num_blocks, block_size >>> ( static_cast<col_type*>(concat_latitudes->data),
+                        static_cast<col_type*>(concat_longitudes->data), offset_size_polygons,
+                        num_polygons, static_cast<col_type*>(perimeter_output.data) );
+        }
+        
+        CHECK_STREAM(stream);
+
+        return perimeter_output;
+    }
+
+    template <typename col_type, std::enable_if_t< !is_supported<col_type>() >* = nullptr>
+    gdf_column operator()( gdf_column* polygons_latitudes[], gdf_column* polygons_longitudes[],
+                           gdf_size_type* const vector_offset_data, gdf_size_type const num_polygons, 
+                           gdf_size_type const offset_values, cudaStream_t stream = 0 )
+    {
+        CUDF_FAIL("Non-arithmetic operation is not supported");
+    }
+};
+    
+}   //  namespace
+ 
+namespace cudf {
+namespace gis {
+
+gdf_column perimeter( gdf_column* polygons_latitudes[], gdf_column* polygons_longitudes[],
```

Comment:
Can you explain the reason that the interface takes an array of columns? Since you have to concatenate them all (not a free operation when there is a valid bitmask), might it be better for the user to provide concatenated columns and an array of polygon lengths? 

I admit I don't know the details of your use case but it seems if you have a lot of small polygons using a column per polygon would be inefficient.

Reply:
For the user it's more easy to have multiple polygons and simply run that. I don't know if there is a specific use case but  it might have support. If you think better the user pass concatenated columns and an array of polygon lengths, I could change that.